internal/config/config.go

```diff
@@ -96,6 +97,18 @@ type HCCMConfiguration struct {
 	Route        RouteConfiguration
 }
 
+func readFromEnvOrFile(envVar string) (string, error) {
+	value := os.Getenv(envVar)
+	if strings.HasPrefix(value, "file:") {
+		valueBytes, err := os.ReadFile(strings.TrimPrefix(value, "file:"))
+		if err != nil {
+			return "", errors.New("failed to read " + envVar + " from file: " + err.Error())
```

Comment:
```suggestion
			return "", fmt.Errorf("failed to read %s from file: %w", envVar, err) 
```

I'd recommend wrapping the underlying error, and leverage `fmt.Errorf`.